### PR TITLE
Support for Authenticate

### PR DESCRIPTION
**Pull request by ben-hypermint**
- Added support for Authenticate product
- Enabled fetching contract information by from HM servers

### CODE REVIEW

File: src/authenticate/Authenticate.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    private wallet: Wallet;
+
+    constructor(private _config: AuthenticateConfig) {
+        this.logger.setConfig(_config);
+
+        this.wallet = new Wallet({
+            ..._config,
+
+            // Restrictions are applied on the server so these values can be defaulted
+            networkEnvironment: NetworkEnvironment.Mainnet,
```
Comment (jono-hypermint) [inline src/authenticate/Authenticate.ts:18]: how do we test this? Might be missing the point

File: src/contracts/EVMContract.ts [hunk 2]
```diff
@@ -670,6 +680,23 @@ export class EVMContract extends BaseContract implements IContract {
...
+        const { contractAddress, contractType, chain, type, environment } =
+            network;
+
+        this.config = {
+            ...this.config,
+            contractAddress,
+            contractType,
+            networkChain: chain,
+            networkType: type,
+            networkEnvironment: environment
```
Comment (jono-hypermint) [inline src/contracts/EVMContract.ts:694]: ah this is how we test - nevermind

File: src/wallets/Wallet.ts [hunk 3]
```diff
@@ -140,4 +140,12 @@ export class Wallet {
         this._signer = this._web3Provider.getSigner();
         this._logger.log('connect', 'Connected');
     }
+
+    public async getAddress(): Promise<string> {
+        return this._signer.getAddress();
```
Comment (jono-hypermint) [inline src/wallets/Wallet.ts:145]: Does this always work? Im 99% sure I had problems with getting the address from the signer (I think in polygon?)

**Review by jono-hypermint — CHANGES_REQUESTED** [review]

File: src/wallets/Wallet.ts (hunk 3 above)
Comment (ben-hypermint) [reply src/wallets/Wallet.ts:141]: Yep. Tested across different networks

**Review by jono-hypermint — APPROVED** [review]

**Pull request merged**